src/TestSuite/Migrator.php

```diff
@@ -0,0 +1,253 @@
+<?php
+declare(strict_types=1);
+
+/**
+ * Copyright (c) Cake Software Foundation, Inc. (http://cakefoundation.org)
+ *
+ * Licensed under The MIT License
+ * Redistributions of files must retain the above copyright notice.
+ *
+ * @copyright     Copyright (c) Cake Software Foundation, Inc. (http://cakefoundation.org)
+ * @link          http://cakephp.org CakePHP(tm) Project
+ * @license       http://www.opensource.org/licenses/mit-license.php MIT License
+ */
+namespace Migrations\TestSuite;
+
+use Cake\Console\ConsoleIo;
+use Cake\Datasource\ConnectionManager;
+use Cake\TestSuite\Schema\SchemaCleaner;
+use Cake\TestSuite\TestConnectionManager;
+use Migrations\Migrations;
+
+class Migrator
+{
+    /**
+     * @var ConfigReader
+     */
+    protected $configReader;
+
+    /**
+     * @var ConsoleIo
+     */
+    protected $io;
+
+    /**
+     * Migrator constructor.
+     * @param bool $verbose
+     * @param null $configReader
+     */
+    final public function __construct(bool $verbose, ?ConfigReader $configReader = null)
+    {
+        $this->io = new ConsoleIo();
+        $this->io->level($verbose ? ConsoleIo::NORMAL : ConsoleIo::QUIET);
+        $this->configReader = $configReader ?? new ConfigReader();
+
+        // Make sure that the connections are aliased, in case
+        // the migrations invoke the table registry.
+        TestConnectionManager::aliasConnections();
+    }
+
+    /**
+     * General command to run before your tests run
+     * E.g. in tests/bootstrap.php
+     *
+     * @param array $config
+     * @param bool  $verbose Set to true to display messages
+     * @return Migrator
+     */
+    public static function migrate(array $config = [], $verbose = false): Migrator
+    {
+        $migrator = new static($verbose);
+
+        $migrator->configReader->readMigrationsInDatasources();
+        $migrator->configReader->readConfig($config);
+        $migrator->handleMigrationsStatus();
+
+        return $migrator;
+    }
+
+    /**
+     * Import the schema from a file, or an array of files.
+     *
+     * @param string $connectionName Connection
+     * @param string|string[] $file File to dump
+     * @param bool $verbose Set to true to display messages
+     * @return void
+     * @throws \Exception if the truncation failed
+     * @throws \RuntimeException if the file could not be processed
+     */
+    public static function dump(string $connectionName, $file, bool $verbose = false)
```

Comment:
This looks a lot like the code we currently have in CakePHP for loading schema from SQL dumps. Do we need it here as well?

Reply:
True, it is not needed anymore!